### PR TITLE
ENT-3422 Add JUnit 5 dependencies to all projects

### PR DESCRIPTION
**Pull request by r3domfox**
JUnit 5 dependencies are added to all projects that use JUnit 4, as a preliminary to switching individual tests, and eventually entire test suites, over to JUnit 5. The `junit-vintage-engine` is used to maintain support for JUnit 4 tests in a mixed JUnit 4/ JUnit 5 environment.

The JUnit 5 version is set to `5.5.0-M1`, as previous versions are prone to a `Malformed Class Name` exception while classpath-scanning Kotlin projects. When the `5.5.x` GA release becomes available, we should upgrade to that.

### CODE REVIEW

File: client/jackson/build.gradle [hunk 1]
```diff
@@ -18,8 +18,16 @@ dependencies {
 
     testCompile project(':test-utils')
     testCompile project(path: ':core', configuration: 'testArtifacts')
+
+    testCompile "org.junit.jupiter:junit-jupiter-api:${junit_jupiter_version}"
+    testCompile "junit:junit:${junit4_version}"
+
+    testRuntime "org.junit.vintage:junit-vintage-engine:${junit_vintage_version}"
```
Comment (chrisr3) [inline client/jackson/build.gradle]: Could all these be `testRuntimeOnly` dependencies?

File: djvm/djvm/build.gradle [hunk 2]
```diff
@@ -34,8 +34,13 @@ dependencies {
     // ClassGraph: classpath scanning
     shadow "io.github.classgraph:classgraph:$class_graph_version"
 
+    testCompile "org.junit.jupiter:junit-jupiter-api:${junit_jupiter_version}"
```
Comment (chrisr3) [inline djvm/djvm/build.gradle]: The DJVM is a separate Gradle project these days and so will not have the `$junit_jupiter_version` property. I've also migrated it not to use the deprecated `testCompile` and `testRuntime` configurations.

**Comment by rick-r3** [comment]
Seems like I don't understand something.  This seems to keep JUnit 4 around and adds JUnit 5.  In my head I was expecting JUnit 5 to have a backwards compatible mode and that JUnit 4 no longer required.

I'm inferring my preconceptions were wrong?

**Comment by r3domfox** [comment]
JUnit 5 maintains a "vintage" engine which identifies and runs JUnit 4 tests. The JUnit 4 dependency provides the `org.junit.Test` annotation, amongst other things, which JUnit 5 does not provide (it uses `org.junit.jupiter.api.Test`). However, it looks like `junit-jupiter-engine` bundles the necessary JUnit 4 pieces, so I've removed the explicit JUnit 4 imports.

**Comment by r3domfox** [comment]
One of the `testing` libs has JUnit 4 as a dependency, so everything that uses that as a `testCompile` dependency gets JUnit 4 transitively. Modules that don't, have to pull in `junit-vintage-engine` as an explicit `testCompile` dependency, rather than `testRuntime`.

File: testing/test-db/gradlew [hunk 3]
```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env sh
```
Comment (rick-r3) [inline testing/test-db/gradlew]: Do these differ in some way from the other `gradlew` ?  Can we add a description of how this file differs or why it exists?

Should they be part of this PR?

File: common/validation/build.gradle [hunk 4]
```diff
@@ -8,6 +8,7 @@ dependencies {
     compile "org.jetbrains.kotlin:kotlin-reflect:$kotlin_version"
 
     testCompile project(":test-utils")
+
```
Comment (rick-r3) [inline common/validation/build.gradle]: Revert this file ?

File: build.gradle [hunk 5]
```diff
@@ -39,6 +39,12 @@ buildscript {
     ext.typesafe_config_version = constants.getProperty("typesafeConfigVersion")
     ext.fileupload_version = '1.3.3'
     ext.junit_version = '4.12'
```
Comment (rick-r3) [inline build.gradle]: Perhaps worth a comment or something to imply this is legacy.

File: testing/test-db/gradlew (hunk 3 above)
Comment (r3domfox) [reply testing/test-db/gradlew]: Nope, accidental commit. Removed it.

File: common/validation/build.gradle (hunk 4 above)
Comment (r3domfox) [reply common/validation/build.gradle]: Done.

File: build.gradle (hunk 5 above)
Comment (r3domfox) [reply build.gradle]: Done.

File: client/jackson/build.gradle (hunk 1 above)
Comment (r3domfox) [reply client/jackson/build.gradle]: These two in particular can't (we need the JUnit 4 from `junit-vintage-engine`, and the JUnit 5 annotations etc, at test compile time)

**Pull request merged**